### PR TITLE
Carry on commits from #2033

### PR DESCRIPTION
**Pull request by dmp42**
A couple of small cleanup and enhancements that were dropped after the revert.

### CODE REVIEW

**Comment by dgageot** [comment]
@dmp42 It looks like GetIP() is called only by the test. Am I right?

**Comment by dgageot** [comment]
@dmp42 Also, you could use that commit to fix `GetMachineName()`'s comment :-)

**Comment by dmp42** [comment]
@dgageot fixed the comment and removed duplicated GetIP definitions.
PTAL

**Comment by dgageot** [comment]
LGTM

**Comment by hairyhenderson** [comment]
Nice. :+1:

File: drivers/generic/generic.go [hunk 1]
```diff
@@ -16,17 +18,15 @@ import (
 
 type Driver struct {
 	*drivers.BaseDriver
-	SSHKey string
```
Comment (nathanleclaire) [inline drivers/generic/generic.go]: Changing this field name _will_ be problematic without adding a `json:"SSHKey"` tag to indicate that in marshalling / demarshaling it should look for that field name.  Existing `config.json` for `generic` machines will break trying to look for that field.  My feeling on the matter is largely that the benefit of changing such a field name does not outweight the cost.

File: drivers/generic/generic.go [hunk 2]
```diff
@@ -83,33 +81,30 @@ func (d *Driver) GetSSHUsername() string {
 func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
 	d.IPAddress = flags.String("generic-ip-address")
 	d.SSHUser = flags.String("generic-ssh-user")
-	d.SSHKey = flags.String("generic-ssh-key")
+	d.sourceSSHKey = flags.String("generic-ssh-key")
 	d.SSHPort = flags.Int("generic-ssh-port")
 
 	if d.IPAddress == "" {
-		return fmt.Errorf("generic driver requires the --generic-ip-address option")
+		return errors.New("generic driver requires the --generic-ip-address option")
```
Comment (nathanleclaire) [inline drivers/generic/generic.go]: If you're going to be changing these to `errors.New` anyway, looks like a good candidate for a custom `ErrRequiredFlagMissing` type.

Comment (dmp42) [inline drivers/generic/generic.go]: Agreed. That can happen over all the codebase as a follow-up PR.

**Comment by dmp42** [comment]
Reverted visibility change on the SSHKey in the driver interface.

**Comment by dmp42** [comment]
PTAL

File: drivers/generic/generic.go (hunk 1 above)
Comment (dgageot) [inline drivers/generic/generic.go]: Good catch @nathanleclaire 
Shouldn't we have a test to protect us from this kind of issue?

**Comment by nathanleclaire** [comment]
LGTM

**Pull request merged**